### DISCUSSION TITLE
Database queries take too long when SQL error is thrown graceful error

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [livewire/livewire#8414] [7. Bugs (archived)]
### Livewire version

3.4.12

### Laravel version

11.7.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

When performing a database query via the query builder on a MySQL 8 database Livewire seems to quietly handle MySQL errors, skip them and seems to dismantle and reconstruct the query so that it works? For an example, running the following query straight on my database through a client like Sequel Ace throws an error on the `birthday` field because birthday is a `date` column, so I instantly get the error:

```sql
select
  applications.*,
  (
    select
      count(*)
    from
      application_entries
    where
      applications.id = application_entries.application_id
  ) as entries_count
from
  applications
where
  (
    id = 'test'
    or hash = 'test'
    or email = 'test'
    or mobile = 'test'
    or birthday = 'test'
  )
order by
  created_at desc
limit
  11 offset 0
```

But when running through the query builder via Livewire, the query runs for over 90 seconds and then returns results, it feels like something magical is being handled directly in Livewire here?

### Repository or Wirebox URL

https://github.com/livewire/livewire/discussions/8407

### Additional information

Previously opened in discussions, for reference the query builder code in my Livewire page is:

```php
<?php

namespace App\Livewire\Pages\Applications;

use Livewire\Attributes\Url;
use Illuminate\Support\Str;
use Livewire\WithPagination;
use Livewire\Component;
use App\Models\Application;

class ApplicationIndex extends Component
{
    use WithPagination;

    #[Url]
    public string $search = '';
    public string $sortField = 'created_at';
    public string $sortDirection = 'desc';
    public bool $showAdvancedFilters = false;
    public array $filters = [
        'product' => null
    ];

    public function sortBy(string $field)
    {
        $this->sortDirection = $this->sortField === $field
            ? $this->sortDirection = $this->sortDirection == 'asc' ? 'desc' : 'asc'
            : 'asc';

        $this->sortField = $field;
    }

    public function updatedSearch()
    {
        $this->resetPage();
    }

    public function updatedFilters()
    {
        $this->resetPage();
    }

    public function render()
    {
        $this->authorize('viewAny', Application::class);

        $applications = Application::withCount('entries')->with([
            'entry'
        ])->when($this->filters['product'], fn ($query, $product) => $query->where('product_type', $product))
            ->whereExact(['id', 'hash', 'email', 'mobile', 'birthday'], $this->search)
            ->orderBy($this->sortField, $this->sortDirection)
            ->simplePaginate(10);

        return view('livewire.pages.applications.application-index', [
            'applications' => $applications
        ]);
    }
}
```

And `searchExact` is:

```php
Builder::macro('searchExact', function (array $fields, $string) {
    return $string
        ? $this->whereAny($fields, '=', Str::of($string)->trim()->toString())
        : $this;
});
```

So for now I've removed `birthday` as it feels like this needs to be handled using a carbon parse?

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PerryvanderMeer** [comment]
Can you please try to execute the Eloquent query outside of Livewire to see if it's related to Livewire?
Besides that, if you dump the generated SQL with `$applications->dumpRawSql()`, does it match your expectations?

**Comment by NBA707** [comment]
I don't see a whereExact statement in Laravel?   When i do this Exact is a field name in the database so it gets translated to where('Exact', ?)   and ? is equal to the first value of the array you are passing.

**Comment by NBA707** [comment]
You can debug it with:

`\DB::enableQueryLog(); // Enable query log

// Your Eloquent query executed by using get()

dd(\DB::getQueryLog()); // Show results of log`
